apps/labs/posts/numba-polynomial-support.md

```diff
@@ -0,0 +1,64 @@
+
+After my first Pull Request was merged, I moved on to other functions from the list. Other NumPy functions that I added support as a Quansight Labs intern include `np.vsplit`, `np.hsplit`, `np.dsplit`, `np.rowstack`, `np.diagflat`, `np.resize`, `np.indices`, `np.unwrap`. A great guide to using the `@overload` decorator is given in the [Numba documentation](https://numba.readthedocs.io/en/stable/extending/overloading-guide.html#a-guide-to-using-overload).
+
+## The NumPy polynomial API
+When choosing functions to implement, I was always looking for groups of similar functions. That way, I could get more functions checked from the list more quickly. This led me to implementing `polyadd, polysub, polymul` from `numpy.polynomial.polynomial` simultaneously. After seeing this, my mentor asked me if I would like to focus on support for the NumPy polynomial package. I agreed since I felt like this would be a good way of working with lower levels of Numba. In addition to adding other functions from the polynomial API, the biggest task during my internship was implementing the `Polynomial` class. It is fairly simple as it has three parameters - `coef`, `domain`, and `window`, each of which is a NumPy array. This meant that the foundation of the implementation was fairly straightforward. Unfortunately, there were a lot of details in the behaviour of the class, and each small detail seemed challenging when I had to implement it using the low-level API. The most interesting thing about the implementation of the class is that the two optional parameters - `domain` and `window`, have default values that are NumPy arrays of integer dtype, whereas if the user decides to set `domain` and `window`, then these arrays are always cast to arrays of double dtype. This meant that the user could never set the values of these two parameters to their default values.
+
+## Adding support for Power Series in Numba
+Similarly to functions, NumPy classes have to be reimplemented in Numba using its low-level API. A good example of this can be found in the [Numba documentation](https://numba.readthedocs.io/en/stable/extending/interval-example.html#example-an-interval-type), which I followed very closely. The first big task when working on the `Polynomial` class was defining a new Numba type to represent the class. After creating a type class called `PolynomialType`, I needed to "show" Numba that each instance of the `Polynomial` class should be treated by Numba as if its type is `PolynomialType`, using the `@typeof_impl.register(np.polynomial.polynomial.Polynomial)` decorator. Another important part is defining the constructor for the Polynomial class. Me and my mentor decided to add support for `Polynomial(coef)` and `Polynomial(coef, domain, window)` at this point. As a general rule, nopython mode does not work on Python objects directly. Each type supported in nopython mode therefore has to define a tailored native representation, also called a data model.
```

Comment:
```suggestion
Similarly to functions, NumPy classes have to be reimplemented in Numba using its low-level API. A good example of this can be found in the [Numba documentation](https://numba.readthedocs.io/en/stable/extending/interval-example.html#example-an-interval-type), which I followed very closely. The first big task when working on the `Polynomial` class was defining a new Numba type to represent the class. After creating a type class called `PolynomialType`, I needed to "show" Numba that each instance of the `Polynomial` class should be treated by Numba as if its type is `PolynomialType`, using the `@typeof_impl.register(np.polynomial.polynomial.Polynomial)` decorator. Another important part is defining the constructor for the Polynomial class. We decided to add support for `Polynomial(coef)` and `Polynomial(coef, domain, window)` at this point. The next step is to define a data model for the new type. That is, how the Polynomial type will be represented in LLVM IR. 
```